fenix/app/src/main/java/org/mozilla/fenix/shopping/ui/ProductAnalysis.kt

```diff
@@ -121,6 +150,129 @@ private fun AdjustedProductRatingCard(
     }
 }
 
+@Composable
+private fun HighlightsCard(
+    highlights: Map<HighlightType, List<String>>,
+    modifier: Modifier = Modifier,
+) {
+    ReviewQualityCheckCard(modifier = modifier) {
+        var isExpanded by remember { mutableStateOf(false) }
+        val highlightsForCompactMode = remember(highlights) { highlights.forCompactMode() }
+        val highlightsToDisplay = remember(isExpanded, highlights) {
+            if (isExpanded) {
+                highlights
+            } else {
+                highlightsForCompactMode
+            }
+        }
```

Comment:
👍 Using `remember` to calculate and storing these states for recomposition like this kinda blew my mind. Very cool!

Reply:
Nice! it's good to remember for cpu "intensive" operations as we don't want to run them on each composition. I learnt that from this [talk - this section talks about this exactly](https://youtu.be/6BRlI5zfCCk?t=1318), the whole talk is really insightful about how compose recomposes and the data structure it uses internally.